### DISCUSSION TITLE
Barrel integration

### DISCUSSION THREAD

**Discussion by marknefedov** [SeaQL/sea-orm#316] [Ideas]
Hey  just a shower though. It would be really great to support Barrel https://git.irde.st/spacekookie/barrel for schema generation and Refinery https://github.com/rust-db/refinery for migrations.
That way it will be possible to handle migrations completely from rust code, that can be handy for redistributable apps.
As per integration I mean generating entities and their relations from barrel/refinery schema definitions.

**Comment by tyt2y3** [comment]
Basically we had a plan for developing our own migration tool too, and it will probably happen in 0.5.

We want to maintain the tool ourselves, to provide a more consistent developer experience, and indeed much of the internal implementation can be shared.

I liked barrel indeed! Is there any feature in barrel that might be missing in SeaQuery?

That say, refinery integration might be useful, even if we developed our own migrator.

**Comment by marknefedov** [comment]
Oh, sorry I've somehow overlook the fact that SeaQuery exists it covers all of my needs!
